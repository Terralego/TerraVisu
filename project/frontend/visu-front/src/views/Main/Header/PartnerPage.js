import React from 'react';
import { Card } from '@blueprintjs/core';


const PartnerPage = () => (
  <Card>
    <div className="mentions-legales">
      <h1>Put your partner data here</h1>
    </div>
  </Card>
);

export default PartnerPage;
