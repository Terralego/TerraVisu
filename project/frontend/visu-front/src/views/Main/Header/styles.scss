@import '../../variables.scss';

.main {

  &__header {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
    flex: 0 0 3.35rem;
    z-index: 100;

    &__target {
      position: absolute;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
      z-index: 105;
    }

    &--mobile {
      flex: 0 0 10rem;
      transform: translateX(0rem);
      margin-left: -7.65rem;
      transition: transform .2s ease-in;

      .header-link {
        display: flex;
        flex-direction: row;
        align-items: center;
        align-content: space-between;
        font-size: .7rem;
        text-align: right;
        outline: none;
        text-decoration: none;
        width: 9.5rem;
        height: 3rem;

        .header-button-label {
          color: #f5f8fa;
          display: table-cell;
          padding-right: 1rem;
          width: 10rem;
          margin: 0;
        }
      }

      .header-button {
        width: 2.25rem;
        height: 2.25rem;
        margin: auto;
        align-items: center;
      }

      .active .header-button-label {
        font-weight: 600;
        font-size: .75rem;
      }

      &--open {
        transform: translateX(7.65rem);
      }
    }
  }

  &__content {
    display: flex;
    flex: 1;
    justify-content: center;
    align-items: center;
    min-width: 0;
  }

  @media print {
    display: block;
    &__header {
      display: none;
    }
    &__content {
      display: block;
    }
  }

  &__navbar {
    display: flex;
    flex-direction: column;
    height: auto;
    flex: 1;
    padding: 0;

    .navBar__header,
    .navBar__group {
      display: flex;
      padding: .5rem;
    }
    
    .navBar__header {
      text-align: center;
      margin: 0;

      #welcome img {
        width: 2.4rem;
        height: auto;
      };
    }

    .navBar__header + .navBar__group {
      margin-top: 0;
    }

    .navBar__group {
      padding-top: .2rem;
      margin: auto 0;
      flex-direction: column;
      height: auto;
      &:last-of-type {
        margin-bottom: 0;
      }

      ul {
        list-style: none;
        padding: 0;
      }

      li {
        margin: 0 0 1rem;
      }

      button {
        outline: none;
      }
    }

    .header-button {
      img,
      svg {
        width: 16px;
        transition: .3s ease-in filter;
      }
    }

    .log-out .header-button {
      background: radial-gradient(#5a91b3 0, transparent 70%);
      svg {
        filter: invert(.5) sepia(1) saturate(5) hue-rotate(175deg);
      }
    }

    .navBar__group .active .header-button {
      background: radial-gradient(#5a91b3 0, transparent 55%);
      svg {
        filter: invert(.5) sepia(1) saturate(5) hue-rotate(175deg);
      }
    }

    // Below override blueprint
    .bp3-overlay {
      white-space: nowrap;
    }
  }
}

.modal-signin {
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 130;

  &__form {
    max-width: 30rem;
    min-width: 20rem;

    &-buttons {
      display: flex;
      justify-content: center;

      button {
        margin: .25rem;
      }
    }
  }
}

.modal-mentions-legales {
  display: flex;
  justify-content: center;
  align-items: center;
  border: none;
  z-index: 130;

  .bp3-card {
    padding: 0;
  }

  .mentions-legales {
    color: #394b59;
    padding: 1.5rem;

    .mentions-legales-container {
      font-family: "SFProText-Regular", Arial, Helvetica, sans-serif;
    }

    &-title {
      text-align: center;
      font-size: 1.5rem;
      font-weight: 500;
      font-family: "SFProText-Medium", Arial, Helvetica, sans-serif;
      margin: 0 0 2rem 0;
    }

    .container-block {
      > h4 {
        margin-bottom: 0;
        font-family: "SFProText-Medium", Arial, Helvetica, sans-serif;
        padding-bottom: 0.4rem;
      }

      > address {
        font-style: normal;
      }

      p {
        margin-bottom: 0.2rem;
      }

      a {
        color: #394b59;
      }

      a:hover {
        font-weight: bold;
      }
    }
  }
}

@media(min-width: $breakpoints-tablet) and (max-width: $breakpoints-desktop) and (orientation: landscape) {
  .modal-partner {
    >.bp3-card {
      max-width: 50%;
      min-height: 45%;
    }
  }
}

@media(min-width: $breakpoints-phone) and (max-width: $breakpoints-tablet) and (orientation: portrait) {
  .modal-partner {
    .bp3-overlay-content {
      max-width: 90%;
      max-height: 90%;
      overflow: scroll;
    }
  }
}
