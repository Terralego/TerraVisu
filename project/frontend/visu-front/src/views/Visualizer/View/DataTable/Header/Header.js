import React from 'react';
import classnames from 'classnames';
import { Tag, Intent, Position, Popover, PopoverInteractionKind, Button, AnchorButton } from '@blueprintjs/core';
import ColumnsSelector from '@terralego/core/modules/Table/components/ColumnsSelector';

import Loading from '../../../../../components/Loading';

function getIds (features) {
  return features.map(({ _id }) => _id).join(',');
}

const close = (setLayerState, layer) => () => setLayerState({ layer, state: { table: false } });

export const Header = ({
  loading,
  title,
  resultsTotal,
  toggleExtent,
  extent,
  columns,
  onChange,
  resize,
  full,
  exportData,
  selectedFeatures = [],
  compare = '',
  setLayerState,
  displayedLayer: [layer] = [],
}) => (
  <div
    className={classnames({
      'table-header': true,
      'table-header--is-loading': loading,
    })}
  >
    <div className="table-header__title">
      <p className="table-header__title--label">{title}</p>
      <div className="table-header__counter">
        {loading && (
          <Loading
            className="table-header__loading"
            size={20}
          />
        )}
        <Tag
          className="table-header__count"
          intent={Intent.PRIMARY}
          round
        >
          {resultsTotal} résultat{resultsTotal > 1 ? 's' : ''}
        </Tag>
      </div>
    </div>
    <div className="table-header__actions">
      {exportData && (
        <Popover
          content="Exporter"
          interactionKind={PopoverInteractionKind.HOVER}
        >
          <Button
            onClick={exportData}
            icon="export"
            minimal
            intent={Intent.PRIMARY}
            disabled={!resultsTotal || loading}
          />
        </Popover>
      )}
      {compare && (
        <Popover
          content={
            selectedFeatures.length < 2 || selectedFeatures.length > 3
              ? 'Sélectionnez entre 2 et 3 données pour les comparer'
              : 'Comparer ces données'
          }
          interactionKind={PopoverInteractionKind.HOVER}
        >
          <AnchorButton
            target="_blank"
            href={compare.replace(/\{\{ids\}\}/, getIds(selectedFeatures))}
            icon="comparison"
            minimal
            disabled={selectedFeatures.length < 2 || selectedFeatures.length > 3}
            intent={Intent.PRIMARY}
          />
        </Popover>
      )}
      <Popover
        content={extent ? 'Afficher toutes les données' : 'Filtrer selon l\'emprise de la carte'}
        interactionKind={PopoverInteractionKind.HOVER}
      >
        <Button
          onClick={toggleExtent}
          icon="path-search"
          minimal
          active={extent}
          intent={Intent.PRIMARY}
        />
      </Popover>
      <Popover
        content="Filtrer les propriétés"
        interactionKind={PopoverInteractionKind.HOVER}
      >
        <ColumnsSelector
          columns={columns}
          onChange={onChange}
          position={Position.LEFT}
          locales={{
            displayAllColumns: 'Afficher toutes les colonnes',
            hideAllColumns: 'Cacher toutes les colonnes',
          }}
        />
      </Popover>
      <Popover
        content={full ? 'Réduire' : 'Agrandir'}
        interactionKind={PopoverInteractionKind.HOVER}
      >
        <Button
          onClick={resize}
          icon={full ? 'minimize' : 'maximize'}
          minimal
          active={full}
          intent={Intent.PRIMARY}
        />
      </Popover>

      <Popover
        content="Fermer"
        interactionKind={PopoverInteractionKind.HOVER}
      >
        <Button
          onClick={close(setLayerState, layer)}
          icon="cross"
          minimal
          intent={Intent.PRIMARY}
        />
      </Popover>
    </div>
  </div>
);

export default Header;
