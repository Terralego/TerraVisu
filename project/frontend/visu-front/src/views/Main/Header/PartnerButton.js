import React, { useState } from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';
import {
  Overlay,
  Classes,
} from '@blueprintjs/core';

import withDeviceSize from '@terralego/core/hoc/withDeviceSize';

import NavBarItemDesktop from '@terralego/core/components/NavBarItemDesktop';
import NavBarItemTablet from '@terralego/core/components/NavBarItemTablet';

import PartnerPage from './PartnerPage';

export const PartnerButton = ({ isMobileSized, isPhoneSized, ...props }) => {
  const [isOpen, setOpen] = useState(false);
  const NavBarItem = isMobileSized ? NavBarItemTablet : NavBarItemDesktop;

  return (
    <>
      <NavBarItem {...props} onClick={() => setOpen(true)} />
      <Overlay
        className={classNames(
          Classes.OVERLAY_SCROLL_CONTAINER,
          Classes.LIGHT,
          'modal-mentions-legales',
        )}
        isOpen={isOpen}
        onClose={() => setOpen(false)}
      >
        <div
          className={classNames(
            Classes.CARD,
            Classes.ELEVATION_4,
          )}
        >
          <PartnerPage />
        </div>
      </Overlay>
    </>
  );
};

PartnerButton.propTypes = {
  isMobileSized: PropTypes.bool,
  isPhoneSized: PropTypes.bool,
};

PartnerButton.defaultProps = {
  isMobileSized: false,
  isPhoneSized: false,
};


export default withDeviceSize()(PartnerButton);
