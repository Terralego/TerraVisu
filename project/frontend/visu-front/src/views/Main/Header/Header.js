import React, { useState, useEffect, useCallback, useMemo } from 'react';

import classnames from 'classnames';
import withDeviceSize from '@terralego/core/hoc/withDeviceSize';
import { connectAuthProvider } from '@terralego/core/modules/Auth';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import MainMenu from '@terralego/core/components/MainMenu';
import LoginButton from '@terralego/core/components/LoginButton';
import { fetchAllViews } from '../../../services/visualizer';

import PartnerButton from './PartnerButton';

import './styles.scss';

const getLinkProps = link => (
  !link.startsWith('http') && {
    link: {
      component: NavLink,
      linkProps: {
        hrefAttribute: 'to',
      },
    },
  }
);

export const Header = ({
  env: { VIEW_ROOT_PATH },
  isHeaderOpen,
  isMobileSized,
  toggleHeader,
  authenticated,
  settings: {
    theme: { logo = '', logoUrl = '/' } = {},
    extraMenuItems = [],
    allowUserRegistration,
  },
}) => {
  const [menu, setMenu] = useState([]);
  const { t } = useTranslation();

  const extraMenuItemsToMenu = useMemo(() => extraMenuItems.map(item => (
    { ...item, ...getLinkProps(item.href) }
  )), [extraMenuItems]);

  const generateMenu = useCallback(views => ({
    navHeader: {
      id: 'welcome',
      label: t('menu.home'),
      href: logoUrl,
      icon: logo,
      ...getLinkProps(logoUrl),
    },
    navItems: [
      views,
      extraMenuItemsToMenu,
      [{
        id: 'nav-partenaires',
        component: () => (
          <PartnerButton
            label={t('menu.informations')}
            icon="info-sign"
          />
        ),
      }, {
        id: 'nav-connexion',
        component: () => (
          <LoginButton
            icon={authenticated ? 'log-out' : 'log-in'}
            label={authenticated ? t('menu.logout') : t('menu.login')}
            className={authenticated ? 'log-out' : 'log-in'}
            translate={t}
            allowUserRegistration={allowUserRegistration}
          />
        ),
      }],
    ],
  }), [t, logoUrl, logo, extraMenuItemsToMenu, authenticated, allowUserRegistration]);


  useEffect(() => {
    let isMounted = true;

    const loadViewList = async () => {
      /** Load the view list and add it to base menu only if component mounted */
      const views = await fetchAllViews(VIEW_ROOT_PATH);
      if (!isMounted) return;
      const viewsToMenu = views.map(view => ({ ...view, ...getLinkProps(view.href) }));
      setMenu(generateMenu(viewsToMenu));
    };

    VIEW_ROOT_PATH && loadViewList();

    return () => {
      isMounted = false;
    };
  }, [VIEW_ROOT_PATH, generateMenu]);

  return (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
    <div
      className={classnames(
        'main__header',
        { 'main__header--mobile': isMobileSized },
        { 'main__header--mobile--open': isMobileSized && isHeaderOpen },
      )}
      onClick={() => toggleHeader()}
      role="button"
      tabIndex="-1"
    >
      <MainMenu {...menu} className="main__navbar" />
      {isMobileSized && !isHeaderOpen
        // eslint-disable-next-line jsx-a11y/click-events-have-key-events
        && <div className="main__header__target" role="button" tabIndex="-1" onClick={() => toggleHeader(false)} />
      }
    </div>
  );
};

Header.propTypes = {
  env: PropTypes.shape({
    VIEW_ROOT_PATH: PropTypes.string,
  }),
  isHeaderOpen: PropTypes.bool,
  isMobileSized: PropTypes.bool,
  toggleHeader: PropTypes.func,
  authenticated: PropTypes.bool,
  settings: PropTypes.shape({
    theme: PropTypes.shape({
      logo: PropTypes.string,
      logoUrl: PropTypes.string,
    }),
    extraMenuItems: PropTypes.array,
  }),
};

Header.defaultProps = {
  isHeaderOpen: false,
  isMobileSized: false,
  toggleHeader: () => {},
  authenticated: false,
  settings: {
    theme: {
      logo: '',
      logoUrl: '/',
    },
    extraMenuItems: [],
  },
  env: {
    VIEW_ROOT_PATH: '',
  },
};

export default withDeviceSize()(connectAuthProvider('authenticated')(Header));
