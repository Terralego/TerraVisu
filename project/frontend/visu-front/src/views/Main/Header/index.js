import { connectMain, connectSettings } from '../Provider/context';
import withEnv from '../../../config/withEnv';

import Header from './Header';

export default withEnv(connectMain('isHeaderOpen', 'toggleHeader')(connectSettings('settings')(Header)));
