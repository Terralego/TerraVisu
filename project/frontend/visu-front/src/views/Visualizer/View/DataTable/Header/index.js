import Header from './Header';
import { connectView } from '../../context';

export default connectView(({ setLayerState, layersTreeState }) => ({
  setLayerState,
  displayedLayer: Array
    .from(layersTreeState)
    .find(([, { table }]) => table),
}))(Header);
