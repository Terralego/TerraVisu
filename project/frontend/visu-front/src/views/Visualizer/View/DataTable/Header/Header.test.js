import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';


jest.mock('@terralego/core/modules/Table/components/ColumnsSelector', () =>
  props => (
    <p {...props}>ColumnsSelector</p>
  ));
it('should render correctly', () => {
  const tree = renderer.create((
    <Header
      title="foo"
      resultsTotal={42}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render correctly with no result', () => {
  const tree = renderer.create((
    <Header
      title="foo"
      resultsTotal={1}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
