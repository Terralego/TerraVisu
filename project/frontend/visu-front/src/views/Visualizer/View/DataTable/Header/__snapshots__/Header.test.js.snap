// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="table-header"
>
  <div
    className="table-header__title"
  >
    <p
      className="table-header__title--label"
    >
      foo
    </p>
    <div
      className="table-header__counter"
    >
      <span
        className="bp3-tag bp3-intent-primary bp3-round table-header__count"
      >
        <span
          className="bp3-text-overflow-ellipsis bp3-fill"
        >
          42
           résultat
          s
        </span>
      </span>
    </div>
  </div>
  <div
    className="table-header__actions"
  >
    
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="bp3-button bp3-minimal bp3-intent-primary"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="bp3-icon bp3-icon-path-search"
            icon="path-search"
          >
            <svg
              data-icon="path-search"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                path-search
              </desc>
              <path
                d="M15 14.62l-4-2.4V9.77c-.32.09-.66.15-1 .18v2.27l-4 2.4V8.71c-.38-.31-.72-.66-1-1.06v6.97l-4-2.4V8c.55 0 1-.45 1-1s-.45-1-1-1V1.38l3.15 1.89c.08-.34.18-.66.32-.97L.76.07v.01A.496.496 0 00.5 0C.22 0 0 .22 0 .5v12c0 .18.1.33.25.42v.01l5 3v-.01c.07.05.16.08.25.08s.18-.03.25-.08v.01l4.74-2.85 4.74 2.85v-.01c.09.05.18.08.27.08.28 0 .5-.22.5-.5v-3.78c-.3.17-.63.28-1 .28v2.62zM2 5c0 .55.45 1 1 1s1-.45 1-1-.45-1-1-1-1 .45-1 1zm6-1c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm7.75-.92l-1.19-.72c.18.43.29.9.36 1.38l.08.04v3.39l1 1V3.5c0-.18-.1-.33-.25-.42zM10 2c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm3.3 4.89c.44-.7.7-1.51.7-2.39C14 2.01 11.99 0 9.5 0S5 2.01 5 4.5 7.01 9 9.5 9c.88 0 1.69-.26 2.39-.7l2.41 2.41c.17.18.42.29.7.29a1.003 1.003 0 00.71-1.71l-2.41-2.4zM9.5 8C7.57 8 6 6.43 6 4.5S7.57 1 9.5 1 13 2.57 13 4.5 11.43 8 9.5 8z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <p
          className=""
          locales={
            Object {
              "displayAllColumns": "Afficher toutes les colonnes",
              "hideAllColumns": "Cacher toutes les colonnes",
            }
          }
          position="left"
          tabIndex={0}
        >
          ColumnsSelector
        </p>
      </span>
    </span>
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="bp3-button bp3-minimal bp3-intent-primary"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="bp3-icon bp3-icon-maximize"
            icon="maximize"
          >
            <svg
              data-icon="maximize"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                maximize
              </desc>
              <path
                d="M5.99 8.99c-.28 0-.53.11-.71.29l-3.29 3.29v-1.59c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .55.45 1 1 1h4c.55 0 1-.45 1-1s-.45-1-1-1H3.41L6.7 10.7a1.003 1.003 0 00-.71-1.71zm9-9h-4c-.55 0-1 .45-1 1s.45 1 1 1h1.59l-3.3 3.3a.99.99 0 00-.29.7 1.003 1.003 0 001.71.71l3.29-3.29V5c0 .55.45 1 1 1s1-.45 1-1V1c0-.56-.45-1.01-1-1.01z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="bp3-button bp3-minimal bp3-intent-primary"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="bp3-icon bp3-icon-cross"
            icon="cross"
          >
            <svg
              data-icon="cross"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                cross
              </desc>
              <path
                d="M9.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L6.59 8 3.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
  </div>
</div>
`;

exports[`should render correctly with no result 1`] = `
<div
  className="table-header"
>
  <div
    className="table-header__title"
  >
    <p
      className="table-header__title--label"
    >
      foo
    </p>
    <div
      className="table-header__counter"
    >
      <span
        className="bp3-tag bp3-intent-primary bp3-round table-header__count"
      >
        <span
          className="bp3-text-overflow-ellipsis bp3-fill"
        >
          1
           résultat
          
        </span>
      </span>
    </div>
  </div>
  <div
    className="table-header__actions"
  >
    
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="bp3-button bp3-minimal bp3-intent-primary"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="bp3-icon bp3-icon-path-search"
            icon="path-search"
          >
            <svg
              data-icon="path-search"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                path-search
              </desc>
              <path
                d="M15 14.62l-4-2.4V9.77c-.32.09-.66.15-1 .18v2.27l-4 2.4V8.71c-.38-.31-.72-.66-1-1.06v6.97l-4-2.4V8c.55 0 1-.45 1-1s-.45-1-1-1V1.38l3.15 1.89c.08-.34.18-.66.32-.97L.76.07v.01A.496.496 0 00.5 0C.22 0 0 .22 0 .5v12c0 .18.1.33.25.42v.01l5 3v-.01c.07.05.16.08.25.08s.18-.03.25-.08v.01l4.74-2.85 4.74 2.85v-.01c.09.05.18.08.27.08.28 0 .5-.22.5-.5v-3.78c-.3.17-.63.28-1 .28v2.62zM2 5c0 .55.45 1 1 1s1-.45 1-1-.45-1-1-1-1 .45-1 1zm6-1c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm7.75-.92l-1.19-.72c.18.43.29.9.36 1.38l.08.04v3.39l1 1V3.5c0-.18-.1-.33-.25-.42zM10 2c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm3.3 4.89c.44-.7.7-1.51.7-2.39C14 2.01 11.99 0 9.5 0S5 2.01 5 4.5 7.01 9 9.5 9c.88 0 1.69-.26 2.39-.7l2.41 2.41c.17.18.42.29.7.29a1.003 1.003 0 00.71-1.71l-2.41-2.4zM9.5 8C7.57 8 6 6.43 6 4.5S7.57 1 9.5 1 13 2.57 13 4.5 11.43 8 9.5 8z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <p
          className=""
          locales={
            Object {
              "displayAllColumns": "Afficher toutes les colonnes",
              "hideAllColumns": "Cacher toutes les colonnes",
            }
          }
          position="left"
          tabIndex={0}
        >
          ColumnsSelector
        </p>
      </span>
    </span>
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="bp3-button bp3-minimal bp3-intent-primary"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="bp3-icon bp3-icon-maximize"
            icon="maximize"
          >
            <svg
              data-icon="maximize"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                maximize
              </desc>
              <path
                d="M5.99 8.99c-.28 0-.53.11-.71.29l-3.29 3.29v-1.59c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .55.45 1 1 1h4c.55 0 1-.45 1-1s-.45-1-1-1H3.41L6.7 10.7a1.003 1.003 0 00-.71-1.71zm9-9h-4c-.55 0-1 .45-1 1s.45 1 1 1h1.59l-3.3 3.3a.99.99 0 00-.29.7 1.003 1.003 0 001.71.71l3.29-3.29V5c0 .55.45 1 1 1s1-.45 1-1V1c0-.56-.45-1.01-1-1.01z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
    <span
      className="bp3-popover-wrapper"
    >
      <span
        className="bp3-popover-target"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="bp3-button bp3-minimal bp3-intent-primary"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="bp3-icon bp3-icon-cross"
            icon="cross"
          >
            <svg
              data-icon="cross"
              height={16}
              viewBox="0 0 16 16"
              width={16}
            >
              <desc>
                cross
              </desc>
              <path
                d="M9.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L8 6.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42L6.59 8 3.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 9.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L9.41 8z"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
  </div>
</div>
`;
